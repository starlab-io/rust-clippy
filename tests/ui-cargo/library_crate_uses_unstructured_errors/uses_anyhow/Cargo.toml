[package]
name = "uses_anyhow"
version = "0.1.0"
edition = "2021"
publish = false


[dependencies]
anyhow = "1"