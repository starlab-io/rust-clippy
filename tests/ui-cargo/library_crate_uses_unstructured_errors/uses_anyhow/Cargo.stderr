error: this is an unstructured error type
 --> src/lib.rs:3:40
  |
3 | pub fn uses_anyhow_error_directly() -> Result<(), anyhow::Error> {
  |                                        ^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: try using an error enum
  = note: `-D clippy::library-crates-structured-errors` implied by `-D warnings`
  = help: to override `-D warnings` add `#[allow(clippy::library_crates_structured_errors)]`

error: this is an unstructured error type
 --> src/lib.rs:7:42
  |
7 | pub fn uses_anyhow_error_indirectly() -> anyhow::Result<()> {
  |                                          ^^^^^^^^^^^^^^^^^^
  |
  = note: try using an error enum

error: could not compile `uses_anyhow` (lib) due to 2 previous errors
